Sources/Instrumentation/URLSession/URLSessionInstrumentation.swift

```diff
@@ -549,22 +549,27 @@ public class URLSessionInstrumentation {
 
     private func urlSession(_ session: URLSession, task: URLSessionTask, didFinishCollecting metrics: URLSessionTaskMetrics) {
         let taskId = self.idKeyForTask(task)
-        if (self.requestMap[taskId]?.request) != nil {
-            /// Code for instrumenting colletion should be written here
-            var requestState: NetworkRequestState?
-            queue.sync {
-                requestState = requestMap[taskId]
-                if requestState != nil {
-                    requestMap[taskId] = nil
-                }
-            }
-            if let error = task.error {
-                let status = (task.response as? HTTPURLResponse)?.statusCode ?? 0
-                URLSessionLogger.logError(error, dataOrFile: requestState?.dataProcessed, statusCode: status, instrumentation: self, sessionTaskId: taskId)
-            } else if let response = task.response {
-                URLSessionLogger.logResponse(response, dataOrFile: requestState?.dataProcessed, instrumentation: self, sessionTaskId: taskId)
+
+        var requestState: NetworkRequestState?
+        queue.sync {
+            requestState = requestMap[taskId]
+
+            if requestState?.request != nil {
+                requestMap[taskId] = nil
```

Comment:
Similarly to other methods, except an extra check on the `request`:
https://github.com/open-telemetry/opentelemetry-swift/blob/main/Sources/Instrumentation/URLSession/URLSessionInstrumentation.swift#L530-L536